### PR TITLE
Adding help and tab functions for db_nmap

### PR DESCRIPTION
**Pull request by nstarke**
These functions address certain problems listed in GitHub issue #4353, but do not address all issues in that ticket.  Most notably, this commit adds basic tab completion for db_nmap. Sorry about the commit message coming across as one line.  Please let me know how this should be adjusted so that it can be landed; I am sure there will need to be changes.

### CODE REVIEW

File: lib/msf/ui/console/command_dispatcher/db.rb [hunk 1]
```diff
@@ -1594,11 +1594,19 @@ def cmd_db_nmap(*args)
       print_status("Usage: db_nmap [nmap options]")
       return
     end
-
+
```
Comment (wvu-r7) [inline lib/msf/ui/console/command_dispatcher/db.rb]: Looks like some erroneous whitespace. :)

Comment (nstarke) [inline lib/msf/ui/console/command_dispatcher/db.rb]: I'm planning on removing this line, but I was hoping to do so with any other changes that may be necessary.  As such, I'll keep this on hold until I receive further notice on changes for this PR.  Thanks!

Comment (wvu-r7) [inline lib/msf/ui/console/command_dispatcher/db.rb]: Okay. Will review today. Thanks, @nstarke. :)

Comment (nstarke) [inline lib/msf/ui/console/command_dispatcher/db.rb]: My pleasure sir.

File: lib/msf/ui/console/command_dispatcher/db.rb [hunk 2]
```diff
@@ -1662,6 +1670,45 @@ def cmd_db_nmap(*args)
...
+        tabs.push(out_line.strip.split(':').first)
+      end
+    end
+
+    stderr.each_line do |err_line|
+      next if err_line.strip.empty?
+      print_error(err_line.strip)
+    end
+
+    return tabs
```
Comment (wvu-r7) [inline lib/msf/ui/console/command_dispatcher/db.rb]: Just ```tabs```.

**Comment by wvu-r7** [comment]
Hey, can you amend the commit message? Looks like you missed a newline, so it's all on one line.

**Comment by wvu-r7** [comment]
Looks like this, btw:

```
commit 10811febc6fc6362c6acc66835f184e324496fdd
Author: nstarke <starke.nicholas@gmail.com>
Date:   Thu Mar 12 23:38:01 2015 +0000

    Adding help and tab functions for db_nmap      These functions address certain problems    listed in GitHub issue #4353, but do not       address all issues in that ticket.  Most          notably, this commit adds basic tab     completion for db_nmap.
```

File: lib/msf/ui/console/command_dispatcher/db.rb [hunk 3]
```diff
@@ -1662,6 +1670,45 @@ def cmd_db_nmap(*args)
...
+  def cmd_db_nmap_help
+    nmap =
+        Rex::FileUtils.find_full_path('nmap') ||
+        Rex::FileUtils.find_full_path('nmap.exe')
+
+    stdout, stderr = Open3.capture3([nmap, 'nmap'], '--help')
+
+    stdout.each_line do |out_line|
+      next if out_line.strip.empty?
+      print_status(out_line.strip);
```
Comment (wvu-r7) [inline lib/msf/ui/console/command_dispatcher/db.rb]: Oh, looks like you've been writing C? :P

Comment (nstarke) [inline lib/msf/ui/console/command_dispatcher/db.rb]: Yes I have, as a matter of fact.  Related: I probably won't get around to the requested changes until this weekend.  Sorry for the delay!

File: lib/msf/ui/console/command_dispatcher/db.rb [hunk 4]
```diff
@@ -1629,7 +1637,7 @@ def cmd_db_nmap(*args)
     end
 
     begin
-      nmap_pipe = ::Open3::popen3([nmap, "nmap"], *args)
+      nmap_pipe = ::Open3::popen3([nmap, "nmap"], arguments)
```
Comment (wvu-r7) [inline lib/msf/ui/console/command_dispatcher/db.rb]: Single quotes here for consistency.

File: lib/msf/ui/console/command_dispatcher/db.rb (hunk 3 above)
Comment (wvu-r7) [inline lib/msf/ui/console/command_dispatcher/db.rb]: No problem. I think I'm done with review. Will test once you've pushed up the changes. :)

**Comment by nstarke** [comment]
Sorry for the delay on this! I've updated the PR to address the issues mentioned above.  Please let me know if there are any other stylistic changes deemed necessary!

**Comment by wvu-r7** [comment]
Thanks!!

File: lib/msf/ui/console/command_dispatcher/db.rb (hunk 3 above)
Comment (wvu-r7) [inline lib/msf/ui/console/command_dispatcher/db.rb]: Ugh, still missed the semicolon.

**Comment by wvu-r7** [comment]
@nstarke: You missed one: https://github.com/rapid7/metasploit-framework/commit/8165ae35d0923db0487731b5840aa38b412495c0. Fixed and pushed. :)

**Pull request merged**